components/datetime/src/time_zone.rs

```diff
@@ -269,12 +271,110 @@ impl<'data> TimeZoneFormat<'data> {
         Ok(time_zone_format)
     }
 
+    pub fn try_new_with_config<L, ZP>(
```

Comment:
> nitpick

We should probably rename this to `try_from_config()`. This is a more common naming convention style in Rust.